linode/manager
# M3-5081: Convert TagsInput.tsx to function component & don't request tags for restricted users

## Description
Following up on https://github.com/linode/manager/pull/7582 – convert file to a function component & change the approach from intercepting the error when a user is restricted to not requesting tags for restricted users per https://github.com/linode/manager/pull/7582#pullrequestreview-636619551.

## How to test
Please ensure the tags functionality is unchanged.

You can have an unrestricted user and a restricted user that has access to create Linodes. In the Create flow for the restricted user, no error should display on the Tags field, and there should be no network request to `/tags`.
